src/health_checker.cc

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+#include "health_checker.h"
+
+#include <boost/filesystem.hpp>
+#include <fstream>
+#include <sstream>
+
+namespace google {
+
+HealthChecker::HealthChecker(const MetadataAgentConfiguration& config)
+    : config_(config) {
+  InitialCreation();
+  InitialCleanup();
+}
+
+void HealthChecker::SetUnhealthy(const std::string& state_name) {
+  if (health_files_.count(state_name) == 0) {
+    health_files_[state_name].reset(
```

Comment:
[Optional] Might be a tiny bit more efficient to use [`emplace`](http://en.cppreference.com/w/cpp/container/map/emplace):
```cpp
    health_files_.emplace(state_name,
                          new FileWrapper(config_.HealthCheckFileDirectory(), state_name));
```